### PR TITLE
[Card] Fix padding in small devices

### PR DESCRIPTION
**Pull request by amireynoso**
### WHY are these changes introduced?

Resolves https://github.com/Shopify/polaris-react-deprecated/issues/1999

Card is supposed to have a 16px padding in small devices, as defined in the UI kit. This was causing inconsistencies when Card is used with other components that respect the 16px spacing rule [see [issue](https://github.com/Shopify/polaris-react-deprecated/issues/1999) for screenshots and live demo].

### WHAT is this pull request doing?

Sets the current spacing rules (20px) as desktop rules using the `page-content-when-not-fully-condensed` breakpoint (already in place for Card). And sets 16px as the default value.

#### Questions / concerns

I'm using the `page-content-when-not-fully-condensed` mixin, which was already being used in Card (breakpoint is at around 490px).

However, it's not consistent to other breakpoints in components that should behave similarly, such as ResourceList (breakpoint is at 458px).

Is `page-content-when-not-fully-condensed` the right call in this case? Or should I use the same hard-coded value as ResourceList?

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

#### Before fix
_Notice the difference between the ResourceList spacing in the second card and the card title (mobile only)._

image

#### After fix

_Desktop stays the same._
image

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import * as React from 'react';
import {
  Page,
  Card,
  ResourceList,
  CalloutCard,
  TextStyle,
} from '@shopify/polaris';

export default class Playground extends React.Component<never, never> {
  render() {
    return (
      <Page title="Playground">
        <Card title="Online store dashboard">
          <Card.Section title="Reports">
            <p>View a summary of your online store’s performance.</p>
          </Card.Section>

          <Card.Section title="Summary">
            <p>
              View a summary of your online store’s performance, including
              sales, visitors, top products, and referrals.
            </p>
          </Card.Section>
        </Card>
        <Card title="Card title">
          <ResourceList
            resourceName={{singular: 'customer', plural: 'customers'}}
            items={[
              {
                id: 341,
                url: 'customers/341',
                name: 'Mae Jemison',
                location: 'Decatur, USA',
              },
              {
                id: 256,
                url: 'customers/256',
                name: 'Ellen Ochoa',
                location: 'Los Angeles, USA',
              },
            ]}
            renderItem={this.renderItem}
          />
        </Card>
        <CalloutCard
          title="Customize the style of your checkout"
          illustration="https://cdn.shopify.com/s/assets/admin/checkout/settings-customizecart-705f57c725ac05be5a34ec20c05b94298cb8afd10aac7bd9c7ad02030f48cfa0.svg"
          primaryAction={{
            content: 'Customize checkout',
            url: 'https://www.shopify.com',
          }}
        >
          <p>Upload your store’s logo, change colors and fonts, and more.</p>
        </CalloutCard>
      </Page>
    );
  }

  private renderItem(item: any) {
    const {id, url, name, location} = item;

    return (
      <ResourceList.Item
        id={id}
        url={url}
        accessibilityLabel={`View details for ${name}`}
      >
        <h3>
          <TextStyle variation="strong">{name}</TextStyle>
        </h3>
        <div>{location}</div>
      </ResourceList.Item>
    );
  }
}
```

</details>

### 🎩 checklist

* [x] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [x] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
* [x] Updated [UNRELEASED.md](https://github.com/Shopify/polaris-react/blob/master/UNRELEASED.md)

### CODE REVIEW

File: src/components/Card/Card.scss [hunk 1]
```diff
@@ -17,11 +17,19 @@
 }
 
 .Header {
-  padding: spacing(loose) spacing(loose) 0;
+  padding: spacing() spacing() 0;
```
Comment (amireynoso) [inline src/components/Card/Card.scss]: On https://github.com/Shopify/polaris-react-deprecated/issues/1999#issuecomment-417046795 it was discussed that side paddings should be 16px, but for top and bottom, should it still be 20px?

Comment (chloerice) [reply src/components/Card/Card.scss]: IMO yes, but tagging in @sarahill 👋

**Comment by amireynoso** [comment]
Thanks, @chloerice!

I'll wait for @sarahill's input into what the top/bottom padding should be, and update if needed.

**Comment by amireynoso** [comment]
Thanks, @sarahill and @chloerice! I've just updated the conflicts in the Changelog.

This is my first PR in Polaris, what are the next steps?

**Comment by chloerice** [comment]
> This is my first PR in Polaris, what are the next steps?

Thanks again, @amireynoso 🎉 You're good to merge once the conflicts are resolved in UNRELEASED (it's an ever-moving target 😣). Ping me if you get any percy snapshot stuff that you're unsure of. (It's kind of cryptic, but for this PR the shift to the left of the red marks is expected)

**Comment by amireynoso** [comment]
Hi @chloerice, thank you so much for your help! Seconds before you commented, I resolved the conflicts in the UNRELEASED doc. It made the Percy check fail again. Would you mind re-approving? I'll merge afterwards. Thanks again! 🙌

**Pull request merged**